fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -121,17 +121,13 @@ impl IBitcoindRpc for BitcoinClient {
             .map_err(|error| info!(?error, "Error broadcasting transaction"));
     }
 
-    async fn was_transaction_confirmed_in(
-        &self,
-        transaction: &Transaction,
-        height: u64,
-    ) -> Result<bool> {
-        let block_hash = self.get_block_hash(height).await?;
-        let block = self.0.get_block(&block_hash)?;
-        Ok(block
-            .txdata
-            .iter()
-            .any(|tx| tx.txid() == transaction.txid()))
+    async fn get_tx_block_height(&self, txid: &Txid) -> Result<Option<u64>> {
+        let info = self.0.get_raw_transaction_info(txid, None).ok();
```

Comment:
Are we silently swallowing errors here? Bad for debugging.

Reply:
Yes because this happens when the tx is not found in the mempool which we treat as non-existing.

I opened this issue for ensuring that consensus agrees on UTXOs https://github.com/fedimint/fedimint/issues/2635